pumpkin/src/command/commands/cmd_clear.rs

```diff
@@ -0,0 +1,129 @@
+            inventory.set_slot(slot as u16, None, true).unwrap();
+            target
+                .send_inventory_slot_update(&mut inventory, slot)
+                .await
+                .unwrap();
+        }
+    }
+    items_count
+}
+
+struct ClearExecutor;
+
+#[async_trait]
+impl CommandExecutor for ClearExecutor {
+    async fn execute<'a>(
+        &self,
+        sender: &mut CommandSender<'a>,
+        _server: &crate::server::Server,
+        args: &ConsumedArgs<'a>,
+    ) -> Result<(), InvalidTreeError> {
+        let Some(Arg::Entities(targets)) = args.get(&ARG_TARGET) else {
+            return Err(InvalidConsumptionError(Some(ARG_TARGET.into())));
+        };
+
+        let target_count = targets.len();
+
+        let mut items_count = 0;
+        for target in targets {
+            let item_count = clear_player(target).await;
+            items_count += item_count;
+        }
+
+        let msg = if target_count == 1 {
+            let target = targets.first().unwrap();
+            if items_count == 0 {
+                TextComponent::text_string(format!(
+                    "No items were found on player {}",
+                    target.gameprofile.name
+                ))
+                .color_named(NamedColor::Red)
+            } else {
+                TextComponent::text_string(format!(
+                    "Removed {} item(s) on player {}",
+                    items_count, target.gameprofile.name
+                ))
+            }
+        } else if items_count == 0 {
+            TextComponent::text_string(format!("No items were found on {target_count} players"))
+                .color_named(NamedColor::Red)
+        } else {
+            TextComponent::text_string(format!(
+                "Removed {items_count} item(s) from {target_count} players"
+            ))
+        };
+
+        sender.send_message(msg).await;
+
+        Ok(())
+    }
+}
+
+struct ClearSelfExecutor;
+
+#[async_trait]
+impl CommandExecutor for ClearSelfExecutor {
+    async fn execute<'a>(
+        &self,
+        sender: &mut CommandSender<'a>,
+        _server: &crate::server::Server,
+        _args: &ConsumedArgs<'a>,
+    ) -> Result<(), InvalidTreeError> {
+        let target = sender
+            .as_player()
+            .ok_or(InvalidTreeError::InvalidRequirementError)?;
+
+        let items_count = clear_player(&target).await;
+
+        let msg = if items_count == 0 {
+            TextComponent::text_string(format!(
+                "No items were found on player {}",
+                target.gameprofile.name
+            ))
+            .color_named(NamedColor::Red)
+        } else {
+            TextComponent::text_string(format!(
+                "Removed {} item(s) on player {}",
+                items_count, target.gameprofile.name
+            ))
+        };
+
+        sender.send_message(msg).await;
+
+        Ok(())
+    }
+}
+
+#[allow(clippy::redundant_closure_for_method_calls)] // causes lifetime issues
```

Comment:
I think you should follow the Clippy error. You probably just can replace `&|sender| sender.is_player()`
with `CommandSender::is_player`

Reply:
Using `CommandSender::is_player` here produce an error. The clippy error was when building the message to send to the player.